File: src/lib/types/import.ts
/**
 * Import types - for Obsidian vault import
 */

/** Request to import an Obsidian vault. */
export interface ImportVaultRequest {
  /** Path to the source Obsidian vault. */
  source_path: string;
  /** Optional subfolder within the target vault to import into. */
  target_subfolder: string | null;
}

/** Progress update during vault import. */
export interface ImportProgress {
  /** Current file being processed. */
  current_file: string;
  /** Number of files processed so far. */
  files_processed: number;
  /** Total number of files to process. */
  total_files: number;
  /** Number of properties imported. */
  properties_imported: number;
  /** Number of tags imported. */
  tags_imported: number;
}

/** Result of vault import. */
export interface ImportResult {
  /** Number of notes imported. */
  notes_imported: number;
  /** Number of files copied (includes non-markdown assets). */
  files_copied: number;
  /** Number of properties imported from frontmatter. */
  properties_imported: number;
  /** Number of tags imported (from frontmatter). */
  tags_imported: number;
  /** Duration of import in milliseconds. */
  duration_ms: number;
  /** Any warnings or skipped files. */
  warnings: string[];
}

File: crates/shared_types/src/types/import.rs
//! Import types (for Obsidian vault import).

use serde::{Deserialize, Serialize};
use ts_rs::TS;

/// Request to import an Obsidian vault.
#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[ts(export)]
pub struct ImportVaultRequest {
    /// Path to the source Obsidian vault.
    pub source_path: String,
    /// Optional subfolder within the target vault to import into.
    pub target_subfolder: Option<String>,
}

/// Progress update during vault import.
#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[ts(export)]
pub struct ImportProgress {
    /// Current file being processed.
    pub current_file: String,
    /// Number of files processed so far.
    pub files_processed: i64,
    /// Total number of files to process.
    pub total_files: i64,
    /// Number of properties imported.
    pub properties_imported: i64,
    /// Number of tags imported.
    pub tags_imported: i64,
}

/// Result of vault import.
#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[ts(export)]
pub struct ImportResult {
    /// Number of notes imported.
    pub notes_imported: i64,
    /// Number of files copied (includes non-markdown assets).
    pub files_copied: i64,
    /// Number of properties imported from frontmatter.
    pub properties_imported: i64,
    /// Number of tags imported (from frontmatter).
    pub tags_imported: i64,
    /// Duration of import in milliseconds.
    pub duration_ms: u64,
    /// Any warnings or skipped files.
    pub warnings: Vec<String>,
}
